src/multi/bmp180.cpp

```diff
@@ -0,0 +1,305 @@
+
+  // check calibration register data for errors
+  for (int i = 0; i < calibration_registers_amount; i += 2) {
+    uint16_t verify_word =
+      (calibration_data_buffer[i] << 8) | calibration_data_buffer[i + 1];
+    if (verify_word == 0x0000 || verify_word == 0xFFFF) {
+      hal::safe_throw(hal::io_error(this));
+    }
+  }
+
+  // take the raw calibration data and convert it into the appropriate variables
+  m_calibration_data.ac1 =
+    (calibration_data_buffer[0] << 8) | calibration_data_buffer[1];
+  m_calibration_data.ac2 =
+    (calibration_data_buffer[2] << 8) | calibration_data_buffer[3];
+  m_calibration_data.ac3 =
+    (calibration_data_buffer[4] << 8) | calibration_data_buffer[5];
+  m_calibration_data.ac4 =
+    (calibration_data_buffer[6] << 8) | calibration_data_buffer[7];
+  m_calibration_data.ac5 =
+    (calibration_data_buffer[8] << 8) | calibration_data_buffer[9];
+  m_calibration_data.ac6 =
+    (calibration_data_buffer[10] << 8) | calibration_data_buffer[11];
+  m_calibration_data.b1 =
+    (calibration_data_buffer[12] << 8) | calibration_data_buffer[13];
+  m_calibration_data.b2 =
+    (calibration_data_buffer[14] << 8) | calibration_data_buffer[15];
+  m_calibration_data.mb =
+    (calibration_data_buffer[16] << 8) | calibration_data_buffer[17];
+  m_calibration_data.mc =
+    (calibration_data_buffer[18] << 8) | calibration_data_buffer[19];
+  m_calibration_data.md =
+    (calibration_data_buffer[20] << 8) | calibration_data_buffer[21];
+}
+
+hal::celsius bmp180::temperature()
+{
+  // run helper function which computes the temperature and b5
+  auto temperature_data = compute_temperature_and_b5(m_i2c,
+                                                     m_address,
+                                                     m_calibration_data.ac5,
+                                                     m_calibration_data.ac6,
+                                                     m_calibration_data.mc,
+                                                     m_calibration_data.md);
+
+  // return just the temperature data
+  return temperature_data.temperature;
+}
+
+bmp180::pressure_results bmp180::pressure(int sample_amount)
+{
+  // clamp sample amount to determined maximum
+  sample_amount = std::clamp(sample_amount, 1, m_maximum_samples);
+
+  computation_variables variables;
+  // run helper function which computes the temperature and b5
+  auto temperature_data = compute_temperature_and_b5(m_i2c,
+                                                     m_address,
+                                                     m_calibration_data.ac5,
+                                                     m_calibration_data.ac6,
+                                                     m_calibration_data.mc,
+                                                     m_calibration_data.md);
+
+  // tell sensor to start conversion for pressure
+  hal::byte control_setting =
+    (static_cast<hal::byte>(m_oversampling_setting) << 6) | 0x34;
```

Comment:
```suggestion
  hal::byte control_setting =
    (hal::value(m_oversampling_setting) << 6) | 0x34;
```

Another option is hal::value from the `#include <libhal-util/enum.hpp>` header.